src/transformers/tokenization_utils_base.py

```diff
@@ -2440,6 +2440,7 @@ def prepare_for_model(
         total_len = len_ids + len_pair_ids + (self.num_special_tokens_to_add(pair=pair) if add_special_tokens else 0)
 
         # Truncation: Handle max sequence length
+        overflowing_tokens = []
```

Comment:
Would it make sens to move this allocation after the if statement checking `return_overflowing_tokens` is `True` ?

This would avoid allocating a list if we're not returning overflowing_tokens at all.

Reply:
That would mean it would be overwriting the result of the following statement:

```py
[...]
        total_len = len_ids + len_pair_ids + (self.num_special_tokens_to_add(pair=pair) if add_special_tokens else 0)

        # Truncation: Handle max sequence length
        if truncation_strategy != TruncationStrategy.DO_NOT_TRUNCATE and max_length and total_len > max_length:
            ids, pair_ids, overflowing_tokens = self.truncate_sequences(
                ids,
                pair_ids=pair_ids,
                num_tokens_to_remove=total_len - max_length,
                truncation_strategy=truncation_strategy,
                stride=stride,
            )

        if return_overflowing_tokens:
            overflowing_tokens = []  # Would overwrite the previous result <---------------------
            encoded_inputs["overflowing_tokens"] = overflowing_tokens
            encoded_inputs["num_truncated_tokens"] = total_len - max_length
[...]
```

We could initialize the list inside a `if return_overflowing_tokens` as well. Would you prefer that?

Reply:
What about: 

```python
encoded_inputs.setdefault("overflowing_tokens", []).extend(overflowing_tokens)
```

Here is an example: 

```python
>>> tokens = ["a", "b", "c"]
>>> a = dict()
>>> a.setdefault("overflowing_tokens", []).extend(tokens)
>>> a["overflowing_tokens"]
['a', 'b', 'c']
```

I can't remember if `BatchEncoding` does support `setdefault`